File: frontend/src/types/staffReport.ts
export interface StaffReportStaff {
  id: string;
  first_name: string;
  last_name: string;
  dni: string | null;
  cuil: string | null;
  phone: string | null;
  email: string | null;
  position: string | null;
  specialty: string | null;
  license_number: string | null;
  hire_date: string | null;
  end_date: string | null;
  status: string;
  is_active: boolean;
  notes: string | null;
}

export interface StaffReportShiftAssignment {
  id: string;
  facility_id: string;
  facility_name: string;
  date: string;
  shift_name: string;
  shift_start_time: string;
  shift_end_time: string;
  notes: string | null;
  attendance_status: 'COVERED' | 'INCOMPLETE' | 'NO_RECORD';
  attendance_check_in: string | null;
  attendance_check_out: string | null;
}

export interface StaffReportAttendance {
  id: string;
  facility_id: string;
  facility_name: string;
  check_in: string;
  check_out: string | null;
  notes: string | null;
}

export interface StaffReportSummary {
  total_assignments: number;
  assignments_covered: number;
  assignments_incomplete: number;
  assignments_no_record: number;
  total_attendances: number;
  total_hours: number | null;
}

export interface StaffReportResponse {
  staff: StaffReportStaff;
  from_date: string;
  to_date: string;
  generated_at: string;
  assignments: StaffReportShiftAssignment[];
  attendances: StaffReportAttendance[];
  summary: StaffReportSummary;
}

File: backend/app/schemas/staff_report.py
from pydantic import BaseModel
from uuid import UUID
from datetime import date, datetime, time
from typing import Optional, List


class StaffReportStaff(BaseModel):
    id: UUID
    first_name: str
    last_name: str
    dni: Optional[str]
    cuil: Optional[str]
    phone: Optional[str]
    email: Optional[str]
    position: Optional[str]
    specialty: Optional[str]
    license_number: Optional[str]
    hire_date: Optional[date]
    end_date: Optional[date]
    status: str
    is_active: bool
    notes: Optional[str]


class StaffReportShiftAssignment(BaseModel):
    id: UUID
    facility_id: UUID
    facility_name: str
    date: date
    shift_name: str
    shift_start_time: time
    shift_end_time: time
    notes: Optional[str]
    attendance_status: str
    attendance_check_in: Optional[datetime]
    attendance_check_out: Optional[datetime]


class StaffReportAttendance(BaseModel):
    id: UUID
    facility_id: UUID
    facility_name: str
    check_in: datetime
    check_out: Optional[datetime]
    notes: Optional[str]


class StaffReportSummary(BaseModel):
    total_assignments: int
    assignments_covered: int
    assignments_incomplete: int
    assignments_no_record: int
    total_attendances: int
    total_hours: Optional[float]


class StaffReportResponse(BaseModel):
    staff: StaffReportStaff
    from_date: date
    to_date: date
    generated_at: datetime
    assignments: List[StaffReportShiftAssignment]
    attendances: List[StaffReportAttendance]
    summary: StaffReportSummary
